Off-Top-App/AudioTransformation
# OT-AudioTransformation | Upload Incoming wav files

What I Did:
  1. Migrated the bytes to Wav file format (from Off-Top-Java) to AudioTransformation Microservice.
  2. Created a working Consumer and Producer.
  3. Created a request to upload the audio file to a GCP once it is converted to the .wav format (from bytes).

How To Test:

  1.  To run, **you will need** to download the following JSON file for submission access to the GCP bucket.
       Start by downloading your unique JSON file from this link. (note: each user will have their own key).
       Select the options as shown in the picture.  **Store this JSON file somewhere other than the AudioTransformation       folder**

       https://console.cloud.google.com/apis/credentials/serviceaccountkey?_ga=2.255316360.10588784.1595236130-171369099.1594432070&_gac=1.156622281.1594549487.Cj0KCQjw6ar4BRDnARIsAITGzlCt6nX_EVNK91tpkJ7hptm8xYjdLc2gGGpxvcwL1567bTJDf0rLKqoaAhILEALw_wcB&project=off-top-firebase&folder&organizationId

<img width="595" alt="image" src="https://user-images.githubusercontent.com/51349201/87921523-890fa900-ca2f-11ea-9489-0aa1a6fe3ad5.png">

  2. You will need to create an environment variable named **GOOGLE_APPLICATION_CREDENTIALS**.  This environment variable will store the path of (and have reference to) the JSON file downloaded **in the previous step**.
    -macOS: 
                  Start by entering the vim editor. (if you are using zsh follow these commands exactly)
<img width="450" alt="image" src="https://user-images.githubusercontent.com/51349201/87920804-911b1900-ca2e-11ea-8bd3-92c38f23b59b.png">

                   **Press I** to insert a variable. Enter the absolute path to the JSON file.

<img width="782" alt="Screen Shot 2020-07-20 at 2 25 42 AM" src="https://user-images.githubusercontent.com/51349201/87922336-b3159b00-ca30-11ea-8a3f-e8fef864f399.png">

                  **Press Esc**, then type **:wq** to exit.
                  Save changes by typing **source ~/.zshrc** (if you have bash it will be different).
   -Windows OS:
                 In windows search type environment variable and click on it.
                 Once you are in the menu, under the system variable you are going to create the variable. Here you will include the path to the JSON file we downloaded.
<img width="430" alt="image" src="https://user-images.githubusercontent.com/51349201/87923608-9da17080-ca32-11ea-9767-2abf6f48d0c2.png">


3. Test by having Off-Top-Java, zookeeper, Kafka, and Off-Top-Flutter running.  **Make sure you pull Off-Top Java and Off-Top Flutter prior to testing**.  Record in flutter and check to see if it uploads to the GCP bucket. 
4. (Note: you may need to do a flutter pub get along with making sure all packages are downloaded in Off-Top Java and    AudioTransformation).
